ddtrace/contrib/unittest/patch.py

```diff
@@ -34,30 +50,92 @@ def get_version():
     item._datadog_span = span
 
 
+def _store_module_identifier(test_object):
+    """Store module identifier at `unittest` module instance."""
+    if hasattr(test_object, "test") and hasattr(test_object.test, "_tests"):
+        for module in test_object.test._tests:
+            if len(module._tests) and _extract_module_name_from_module(module):
+                _set_identifier(module, "module")
+
+
+def _store_suite_identifier(module):
+    """Store suite identifier at `unittest` suite instance."""
+    if hasattr(module, "_tests"):
+        for suite in module._tests:
+            if len(suite._tests) and _extract_module_name_from_module(suite):
+                _set_identifier(suite, "suite")
+
+
+def _is_test_module(item):
+    return hasattr(item, "_datadog_object") and item._datadog_object == "module"
+
+
+def _is_test_suite(item):
+    return hasattr(item, "_datadog_object") and item._datadog_object == "suite"
+
+
+def _is_test(item):
+    if type(item) == unittest.TestSuite or not hasattr(item, "_testMethodName"):
+        return False
+    if ddtrace.config.unittest.strict_naming and not item._testMethodName.startswith("test"):
+        return False
+    return True
+
+
 def _extract_span(item):
     """Extract span from `unittest` instance."""
     return getattr(item, "_datadog_span", None)
 
 
-def _extract_test_method_name(item):
+def _extract_command_name_from_session(session):
+    """Extract command name from `unittest` instance."""
+    if not hasattr(session, "progName"):
+        return "python -m unittest"
+    return getattr(session, "progName", None)
+
+
+def _extract_test_method_name(test_object):
     """Extract test method name from `unittest` instance."""
-    return getattr(item, "_testMethodName", None)
+    return getattr(test_object, "_testMethodName", None)
 
 
-def _extract_suite_name_from_test_method(item):
-    item_type = type(item)
-    return getattr(item_type, "__name__", None)
+def _extract_suite_name(item):
+    return type(item._tests[0]).__name__
+
+
+def _extract_session_span():
+    return getattr(_CIVisibility, "_datadog_session_span", None)
+
+
+def _extract_module_span(module_identifier):
+    if hasattr(_CIVisibility, "_unittest_data") and module_identifier in _CIVisibility._unittest_data["modules"]:
+        return _CIVisibility._unittest_data["modules"][module_identifier].get("module_span")
+    return None
+
+
+def _extract_suite_span(suite_identifier):
+    if hasattr(_CIVisibility, "_unittest_data") and suite_identifier in _CIVisibility._unittest_data["suites"]:
+        return _CIVisibility._unittest_data["suites"][suite_identifier].get("suite_span")
+    return None
 
 
-def _extract_class_hierarchy_name(item):
+def _update_status_item(item, status):
+    existing_status = item.get_tag(test.STATUS)
+    if existing_status and (status == test.Status.SKIP.value or existing_status == test.Status.FAIL.value):
+        return
+    item.set_tag_str(test.STATUS, status)
```

Comment:
nit: In some functions we return None and other we don't. We should try to be consistent. 
```suggestion
        return None
    item.set_tag_str(test.STATUS, status)
    return None
```